client/rest/src/main/java/org/elasticsearch/client/RestClient.java

```diff
@@ -796,8 +796,11 @@ static URI buildUri(String pathPrefix, String path, Map<String, String> params)
         Objects.requireNonNull(path, "path must not be null");
         try {
             String fullPath;
-            if (pathPrefix != null) {
-                if (path.startsWith("/")) {
+            if (pathPrefix != null && pathPrefix.isEmpty() == false) {
+                fullPath = (pathPrefix + "/" + path).replaceAll("//+", "/");
```

Comment:
This value is immediately written over in any one of the following `if`, `else if`, or `else` blocks so this line does not do anything?